### PR TITLE
#172101652: Project initial setup

### PR DESCRIPTION
**Pull request by ascii-dev**
## Description
This PR creates the initial setup for the project.

> Maintains #172101652

## Type of change
- [x] Chore (non-breaking change which maintains project)

## Screenshots
Screenshot 2020-04-02 at 9 11 24 AM

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+## COMMONS-UI
```
Comment (karimkawambwa) [inline README.md]: Lets have some description about COMMONS UI

**Review by karimkawambwa — CHANGES_REQUESTED** [review]
While I  do appreciate typescript and what it offers, let's remove it from this PR. We had typescript once...

@ascii-dev to expand on the above it's cool to see you're a TypeScript fan too but we experimented on it and it didn't work for everyone, lets use JS (and babel)

File: README.md (hunk 1 above)
Comment (ascii-dev) [reply README.md]: @karimkawambwa These have been fixed. Kindly help review again. Thanks

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import Demo from "./components/Demo";
+
+export default Demo;
```
Comment (karimkawambwa) [inline src/index.js]: This demo should be in `packages/` directory

**Review by karimkawambwa — COMMENTED** [review]
LGTM 🚀 

Just a change needed as indicated and this should be good to go.

File: src/index.js (hunk 2 above)
Comment (ascii-dev) [reply src/index.js:3]: Thanks @karimkawambwa, one little question though, I should get rid of the `src` folder, right?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+Storybook is an open-source tool that provides a sandbox to develop and visualize components in isolation. See the [Storybook site](https://storybook.js.org/) for more information about the tool. 
+
+COMMONS-UI uses Storybook to view and work with the UI components developed in the COMMONS-UI packages.
+
+Run locally in your development environment running: `yarn start` from the top-level Gutenberg directory.
+
+---
+
+## License
```
Comment (kilemensi) [inline README.md]: Cool, or we can just link to `LICENSE.md` from here

File: package.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
+{
+  "name": "commons-ui",
+  "version": "1.0.0",
```
Comment (kilemensi) [inline package.json]:
```suggestion
```

File: package.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  "description": "Common static components.",
+  "author": "Code for Africa <hello@codeforafrica.org> (https://codeforafrica.org)",
+  "license": "GPL-3.0-or-later",
+  "private": false,
```
Comment (kilemensi) [inline package.json]:
```suggestion
  "private": true,
```

File: package.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    "@storybook/addon-actions": "^5.1.9",
+    "@storybook/addon-console": "^1.1.0",
+    "@storybook/addon-docs": "5.3.2",
+    "@storybook/addon-knobs": "^5.1.9",
+    "@storybook/addon-storyshots": "5.3.2",
+    "@storybook/addon-storysource": "5.3.2",
+    "@storybook/addon-viewport": "5.3.2",
+    "@storybook/addons": "^5.1.9",
+    "@storybook/react": "^5.1.9",
+    "awesome-typescript-loader": "^5.2.1",
```
Comment (kilemensi) [inline package.json]:
```suggestion
```

File: package.json [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    "start": "start-storybook",
+    "build:lib": "yarn copy-svg-to-lib && yarn copy-png-to-lib",
+    "build:storybook": "build-storybook -o docs/",
+    "build": "lerna exec --parallel -- babel --root-mode upward src -d build --copy-files --ignore **/*.story.js,**/*.spec.js",
+    "copy-svg-to-lib": "cpx \"./src/**/*.svg\" ./dist",
+    "copy-png-to-lib": "cpx \"./src/**/*.png\" ./dist"
+  },
+  "dependencies": {
+    "@material-ui/core": "^4.1.3",
+    "@material-ui/types": "^4.1.1",
```
Comment (kilemensi) [inline package.json]:
```suggestion
```

File: package.json [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    "hooks": {
+      "pre-commit": "yarn lint-staged"
+    }
+  },
+  "lint-staged": {
+    "src/**/*.{ts,tsx,json,md}": [
+      "yarn lint-staged",
+      "git add"
+    ]
+  }
```
Comment (kilemensi) [inline package.json]:
```suggestion
  },
    "engines": {
    "node": ">=10.16.0 <=12.x"
  },
  "workspaces": [
    "packages/*"
  ]
```

File: lerna.json [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
+  "packages": [
+    "packages/*"
+  ],
```
Comment (kilemensi) [inline lerna.json]:
```suggestion
  "npmClient": "yarn",
  "useWorkspaces": true,
```

File: packages/README.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+
+### Legacy Patch Release
+
+To release a patch for the older major or minor version of packages, run the following command:
+
+```bash
+npm run publish:packages
+```
+
+This is usually necessary when adding bug fixes or security patches to the earlier versions of HURUmap.
```
Comment (karimkawambwa) [inline packages/README.md]:
```suggestion
This is usually necessary when adding bug fixes or security patches to the earlier versions of Commons-UI.
```

**Review by kilemensi — COMMENTED** [review]
:+1: @ascii-dev ... Not bad for a _backend software engineer_ at all! :clinking_glasses: 

-----------------------------
1. Seems like some dependencies are missing or wrong versions are used. While I agree we shouldn't install stuff like `vue`, we shouldn't be getting warnings about `webpack`, `eslint`, etc.
Screenshot from 2020-04-02 12-51-36
2. Lets start with the most recent versions of all dependencies _unless there are some in incompatibilities_ that prevent us from doing so.
Screenshot from 2020-04-02 12-52-18

File: packages/README.md (hunk 10 above)
Comment (ascii-dev) [reply packages/README.md]: Fixed this... Thanks @karimkawambwa

**Review by nerdlet — COMMENTED** [review]
@ascii-dev Good start :raised_hands: Just remove typescript files from the project and we are good to go.

File: lerna.json [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
```
Comment (karimkawambwa) [inline lerna.json]:
```suggestion
```

**Review by karimkawambwa — APPROVED** [review]
LGTM 🚀

**Review by nerdlet — APPROVED** [review]

File: eslintrc.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
+module.exports = {
```
Comment (kilemensi) [inline eslintrc.js]: Rename file to start with `.`

File: eslintrc.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  extends: [
+    'airbnb',
+    'plugin:prettier/recommended' // Enables eslint-plugin-prettier and displays prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
+  ],
+  plugins: ['react-hooks', 'json', 'markdown'],
+  rules: {
+    'import/no-extraneous-dependencies': [
+      'error',
+      { devDependencies: ['.storybook/**', 'stories/**'] }
+    ],
```
Comment (kilemensi) [inline eslintrc.js]:
```suggestion
    ],
    'react/jsx-filename-extension': [1, { extensions: ['.js'] }],
```

File: package.json [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+    "react": "^16.8.6",
+    "react-dom": "^16.8.6"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "yarn lint-staged"
+    }
+  },
+  "lint-staged": {
+    "src/**/*.{ts,tsx,json,md}": [
```
Comment (kilemensi) [inline package.json]:
```suggestion
    "src/**/*.{js,json,md}": [
```

**Review by kilemensi — APPROVED** [review]
:rocket: @ascii-dev 

----------------------------------

Lets also make sure the `build` folder(s) is/are included in `.gitignore` @ascii-dev .

**Pull request merged**